core/templates/pages/community-dashboard-page/services/contribution-opportunities-backend-api.service.ts

```diff
@@ -17,72 +17,125 @@
  * contributors to contribute.
  */
 
-require('domain/opportunity/ExplorationOpportunitySummaryObjectFactory.ts');
-require('domain/opportunity/SkillOpportunityObjectFactory.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
 
-angular.module('oppia').factory('ContributionOpportunitiesBackendApiService', [
-  '$http', 'ExplorationOpportunitySummaryObjectFactory',
-  'SkillOpportunityObjectFactory', 'UrlInterpolationService',
-  'OPPORTUNITY_TYPE_SKILL', 'OPPORTUNITY_TYPE_TRANSLATION',
-  'OPPORTUNITY_TYPE_VOICEOVER',
-  function($http, ExplorationOpportunitySummaryObjectFactory,
-      SkillOpportunityObjectFactory, UrlInterpolationService,
-      OPPORTUNITY_TYPE_SKILL, OPPORTUNITY_TYPE_TRANSLATION,
-      OPPORTUNITY_TYPE_VOICEOVER) {
-    var urlTemplate = '/opportunitiessummaryhandler/<opportunityType>';
-    var _getOpportunityFromDict = function(opportunityType, opportunityDict) {
-      if (
-        opportunityType === OPPORTUNITY_TYPE_VOICEOVER ||
-        opportunityType === OPPORTUNITY_TYPE_TRANSLATION) {
-        return ExplorationOpportunitySummaryObjectFactory.createFromBackendDict(
-          opportunityDict);
-      } else if (opportunityType === OPPORTUNITY_TYPE_SKILL) {
-        return SkillOpportunityObjectFactory.createFromBackendDict(
-          opportunityDict);
+import { ExplorationOpportunitySummary } from
+  'domain/opportunity/ExplorationOpportunitySummaryObjectFactory';
+import { SkillOpportunity } from
+  'domain/opportunity/SkillOpportunityObjectFactory';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+
+const constants = require('constants.ts');
+
+type OpportunityType = 'skill' | 'voiceover' | 'translation';
+
+// TODO(#7165): Replace any with exact type.
+type ContributionOpportunityParams = {
+  cursor: string;
```

Comment:
I noted the type of cursor below is `number|string` but here it is just `string`. Is everything OK?

Reply:
Since this type is being used in HTTP params they are expected to be of the type string. If we don't specify the type then I think it will convert to string but if I specify the type as number|string it gives me an error saying that HTTP params expect type string or [string].

Reply:
Not a problem anymore as: 
> I just checked the python test file for this API, they are asserting this thing to be a basestring so It is a string, changing it to string in all places.

Reply:
Sg!